### PR TITLE
Proper Promise Implementation

### LINKED ISSUE #7: EventEmitter ≠ Promise
Hi there,

I'm a bit confused with the documentation/readme :

Why would you say [Make the same request with a **promise**](https://github.com/mailjet/mailjet-apiv3-nodejs#make-the-same-request-with-a-promise) or call it a [promise](https://github.com/mailjet/mailjet-apiv3-nodejs/blob/master/mailjet-client.js#L142) when it's clearly [not a promise](https://promisesaplus.com/) ?

I would suggest to either go with promise and use something like [petkaantonov/bluebird](https://github.com/petkaantonov/bluebird) to return a `thenable` or stay with the EventEmitter pattern and stop advertising it as something it isn't.

I've made the required change for a [proper Promise](https://github.com/vizeat/mailjet-apiv3-nodejs/commit/b62fcb6128804c75b5238f8bf99bd31db00c388f?w=1) implementation (both code & test)
along with [consistency fix](https://github.com/mailjet/mailjet-apiv3-nodejs/compare/master...vizeat:feat/promise?expand=1) (code style & line endings). 
I've used [standardjs](http://standardjs.com/) as code style because it is what I use, but if you'd rather keep semicolons you can checkout [semistandard](https://github.com/Flet/semistandard) and LF line endings instead of mixed CR & LF.

If you like it I can make a [pull request](https://github.com/mailjet/mailjet-apiv3-nodejs/compare/master...vizeat:feat/promise?expand=1), I've already bumped the major version in the `package.json` as moving away from EventEmitter is a breaking change.

Either way, thanks for putting together this package as a starting point for using Mailjet API :-)

### PR DESCRIPTION
**Pull request by p-j**
As disccussed, here is an implementation with Promise instead of EventEmitter.

Closes #7

### CODE REVIEW

**Comment by GuillaumeBadi** [comment]
Wow, you even updated the tests! cool ! Can you update the README example just before I merge it ?

**Comment by p-j** [comment]
Sure, just one thing though.
I've used [ES6 arrow functions](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions) and [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) are these ok with you (it basically means that only Node 4 or 5 will run without `--harmony`)

File: test/test.js [hunk 1]
```diff
@@ -1,190 +1,180 @@
...
+const API_KEY = process.env.MJ_APIKEY_PUBLIC
+const API_SECRET = process.env.MJ_APIKEY_PRIVATE
+
+var Mailjet = require('../mailjet-client')
+var chai = require('chai')
+var expect = chai.expect
+var should = chai.should() // eslint-disable-line no-unused-vars
+var Promise = require('bluebird')
+if (typeof API_KEY === 'undefined' || typeof API_SECRET === 'undefined') {
+  throw new Error(`Mailjet API_KEY and API_SECRET are required, respectively "${API_KEY}" and "${API_SECRET}" given`)
```
Comment (p-j) [inline test/test.js]: Template Literals here to exit if mandatory configuration is missing

File: mailjet-client.js [hunk 2]
```diff
@@ -1,375 +1,392 @@
...
+
+  if (this.testMode) {
+    return [options.url, options.body || {}]
+  }
+
+  if (method === 'delete') {
+    method = 'del'
+  }
+
+  return new Promise((resolve, reject) => {
```
Comment (p-j) [inline mailjet-client.js]: ES6 Arrow function here by habit more than anything else.

**Comment by GuillaumeBadi** [comment]
Well, I guess that would be ok to use arrow functions if that is a major release.

**Comment by GuillaumeBadi** [comment]
So this is merged, i will remove the older node versions from travis, then release it on NPM, Thanks for your time !

**Pull request merged**